File: day2/ans.py
def getDigits(n: int) -> list:
    return list(map(int, str(n)))    

class Solution:
    def isHappy(self, n: int) -> bool:
        numbers_processed = {}
        while True:
            digits = getDigits(n)
            sum_square_digits = sum([digit*digit for digit in digits])
            
            if sum_square_digits == 1:
                return True
            
            if sum_square_digits in numbers_processed:
                return False
            
            numbers_processed[n] = 1
            n = sum_square_digits

File: day2/ans.cpp
class Solution {
public:
    bool isHappy(int n) {
        int slow = n;
        int fast = n;
        do {
            slow = digitSquareSum(slow);
            fast = digitSquareSum(digitSquareSum(fast));
            if(fast == 1) {
                return true;
            }
        } while(slow != fast);
        
        return false;
    }

private:
    int digitSquareSum(int n) {
        int sum = 0;
        while(n != 0) {
            int digit = n % 10;
            sum += (digit*digit);
            n /= 10;
        }
        
        return sum;
    }
};